server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -988,6 +986,12 @@ private static void validateSnapshotRestorable(final RepositoryMetadata reposito
                     + "]"
             );
         }
+        if (request.includeGlobalState() && snapshotInfo.includeGlobalState() == Boolean.FALSE) {
+            throw new SnapshotRestoreException(
+                new Snapshot(repository.name(), snapshotInfo.snapshotId()),
+                "the snapshot was created without global state"
```

Comment:
Can we include into the message that we cannot restore with global state? For instance:

```suggestion
                "cannot restore global state since the snapshot was created without global state"
```